### ISSUE TITLE
ms-swift 4.3.1 grpo 模型输出迅速崩塌乱码

### ISSUE THREAD

**Issue by lost-person** [modelscope/ms-swift#9619]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。

### Bug Description / Bug 描述

采用grpo算法 rollout模式 训练 qwen3-32b模型，训练采样completion显示前几步，输出突然崩溃

<img width="2476" height="940" alt="Image" src="https://github.com/user-attachments/assets/c60463b3-8087-4c70-a0ad-9253a59761ea" />

训练端代码如下

```bash
# 项目路径
PROJECT_DIR=$(dirname "$(dirname "$PWD")")
BASE_MODEL_PATH=$1
PLUGIN_PATH=$PROJECT_DIR/plugin/reward.py
DATASET_PATH=$PROJECT_DIR/datasets/grpo/sample.jsonl
OUTPUT_DIR=$PROJECT_DIR/models/grpo/Qwen3-32B/

export NPROC_PER_NODE=8
export VLLM_SERVER_HOST=x.x.x.x
export VLLM_SERVER_PORT=8000

swift rlhf \
    --rlhf_type grpo \
    --tuner_type full \
    --model $BASE_MODEL_PATH \
    --dataset $DATASET_PATH \
    --output_dir $OUTPUT_DIR \
    --load_from_cache_file true \
    --torch_dtype bfloat16 \
    --dataloader_num_workers 8 \
    --max_length 4000 \
    --num_train_epochs 1 \
    --save_total_limit 1 \
    --external_plugins $PLUGIN_PATH \
    --reward_funcs model_reward keyword_reward \
    --reward_weights 1 1 \
    --use_vllm true \
    --vllm_mode server \
    --vllm_server_host $VLLM_SERVER_HOST  \
    --vllm_server_port $VLLM_SERVER_PORT \
    --max_completion_length 1000 \
    --learning_rate 5e-7 \
    --warmup_ratio 0.05 \
    --max_grad_norm 1.0 \
    --beta 0.2 \
    --per_device_train_batch_size 8 \
    --per_device_eval_batch_size 8 \
    --gradient_accumulation_steps 1 \
    --num_generations 4 \
    --eval_steps 100 \
    --save_steps 100 \
    --logging_steps 2 \
    --temperature 0.6 \
    --top_p 0.85 \
    --move_model_batches 128 \
    --deepspeed zero3_offload \
    --attn_impl flash_attn \
    --use_liger_kernel true \
    --log_completions true \
    --early_stop_interval 3
```

推理端代码如下

```bash
# 项目路径
PROJECT_DIR=$(dirname "$(dirname "$PWD")")
BASE_MODEL_PATH=$1

# NODE1 for vLLM Server
swift rollout \
    --model $BASE_MODEL_PATH \
    --vllm_tensor_parallel_size 8 \
    --vllm_gpu_memory_utilization 0.2 \
    --vllm_max_model_len 4000
```

训练日志如下
```json
{"loss": 0.00024637, "grad_norm": 3.83934331, "learning_rate": 1e-08, "completions/mean_length": 36.40625, "completions/min_length": 28.0, "completions/max_length": 54.0, "completions/clipped_ratio": 0.0, "reward": 1.3699975, "reward_std": 0.21771681, "frac_reward_zero_std": 0.0, "rewards/ModelReward/mean": 0.7449975, "rewards/ModelReward/std": 0.11761423, "rewards/KeyWordReward/mean": 0.625, "rewards/KeyWordReward/std": 0.48795006, "kl": 0.00108974, "clip_ratio": 0.0, "step_time": 145.67065608, "epoch": 0.00067751, "global_step/max_steps": "1/1476", "elapsed_time": "2m 29s", "remaining_time": "2d 13h 1m 43s", "memory(GiB)": 19.55, "train_speed(s/it)": 148.951062}
{"loss": 0.00017165, "grad_norm": 2119.7824707, "learning_rate": 1e-08, "completions/mean_length": 34.828125, "completions/min_length": 23.0, "completions/max_length": 50.0, "completions/clipped_ratio": 0.0, "reward": 1.4580704, "reward_std": 0.16815829, "frac_reward_zero_std": 0.0, "rewards/ModelReward/mean": 0.7080704, "rewards/ModelReward/std": 0.14524429, "rewards/KeyWordReward/mean": 0.75, "rewards/KeyWordReward/std": 0.43643579, "kl": 0.00095029, "clip_ratio": 0.0, "step_time": 147.57394905, "epoch": 0.00135501, "global_step/max_steps": "2/1476", "elapsed_time": "4m 57s", "remaining_time": "2d 12h 46m 30s", "memory(GiB)": 23.94, "train_speed(s/it)": 148.432797}
{"loss": 0.0, "grad_norm": 2.23606801, "learning_rate": 3e-08, "completions/mean_length": 995.65625, "completions/min_length": 722.0, "completions/max_length": 1000.0, "completions/clipped_ratio": 0.9921875, "reward": 0.0, "reward_std": 0.0, "frac_reward_zero_std": 1.0, "rewards/ModelReward/mean": 0.0, "rewards/ModelReward/std": 0.0, "rewards/KeyWordReward/mean": 0.0, "rewards/KeyWordReward/std": 0.0, "kl": NaN, "clip_ratio": 0.0, "step_time": 121.47190733, "epoch": 0.00271003, "global_step/max_steps": "4/1476", "elapsed_time": "9m 0s", "remaining_time": "2d 7h 13m 9s", "memory(GiB)": 36.04, "train_speed(s/it)": 135.046699}
```

### How to Reproduce / 如何复现

2台8卡的h800机器 ms-swift 4.3.1 vllm 0.17.0

**Comment by hjh0119** [comment]
run normally on my end, some relevant environment info:
```
vllm                                     0.23.0
transformers                             5.12.0
deepspeed                                0.18.9
liger_kernel                             0.8.0
```

**Comment by lost-person** [comment]
> run normally on my end, some relevant environment info:
> 
> ```
> vllm                                     0.23.0
> transformers                             5.12.0
> deepspeed                                0.18.9
> liger_kernel                             0.8.0
> ```

我debug了下，原因是设置 --use_liger_kernel true，将它删除就解决了该问题。其中，我的版本是
vllm                                     0.17.0
transformers                             5.2.0
deepspeed                                0.18.7
liger_kernel                             0.7.0

**Comment by mccatec** [comment]
ms_swift (not megatron-swift) dev with torch 2.10+ produces NaN logits which cause '!!!!!!' during rollout. Fixed by torch 2.9.1.
